ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -916,15 +916,20 @@ public void forget() throws org.omg.CORBA.SystemException
 		        " _forgotten=" + _forgotten);
 		}
 
-                if (!_forgotten) {
+                if (!_forgotten)
                     handleForget();
 
+                if (_forgotten) {
```

Comment:
(16:58:37) mmusgrov: if _theXAResource is null handleForget just returns and then we used to go ahead anyway and delete the log which is wrong
(17:02:24) mmusgrov: it starts out false and then handleForget() is called. If _theXAResource is null it stays false
(17:03:18) mmusgrov: I can change the signature of private void handleForget() to return a boolean which will make the code easier to parse.